state-chain/pallets/cf-threshold-signature/src/lib.rs

```diff
@@ -1168,31 +1156,41 @@ pub mod pallet {
 			)
 		}
 
-		#[pallet::call_index(7)]
-		#[pallet::weight(T::Weights::set_keygen_response_timeout())]
-		pub fn set_keygen_response_timeout(
+		/// Apply a list of configuration updates to the pallet.
+		///
+		/// Requires Governance.
+		#[pallet::call_index(9)]
+		#[pallet::weight(T::Weights::update_pallet_config())]
+		pub fn update_pallet_config(
 			origin: OriginFor<T>,
-			new_timeout: BlockNumberFor<T>,
+			update: PalletConfigUpdate,
 		) -> DispatchResult {
 			T::EnsureGovernance::ensure_origin(origin)?;
 
-			if new_timeout != KeygenResponseTimeout::<T, I>::get() {
-				KeygenResponseTimeout::<T, I>::put(new_timeout);
-				Pallet::<T, I>::deposit_event(Event::KeygenResponseTimeoutUpdated { new_timeout });
-			}
-
-			Ok(())
-		}
+			match update {
+				PalletConfigUpdate::ThresholdSignatureResponseTimeout { new_timeout } => {
+					let new_timeout: BlockNumberFor<T> = new_timeout.into();
+					if new_timeout != ThresholdSignatureResponseTimeout::<T, I>::get() {
+						ThresholdSignatureResponseTimeout::<T, I>::put(new_timeout);
+						Self::deposit_event(Event::<T, I>::PalletConfigUpdated { update });
+					}
+				},
 
-		#[pallet::call_index(8)]
-		#[pallet::weight(T::Weights::set_keygen_response_timeout())]
-		pub fn set_keygen_slash_amount(
-			origin: OriginFor<T>,
-			amount_to_slash: FlipBalance,
-		) -> DispatchResult {
-			T::EnsureGovernance::ensure_origin(origin)?;
+				PalletConfigUpdate::KeygenResponseTimeout { new_timeout } => {
+					let new_timeout: BlockNumberFor<T> = new_timeout.into();
+					if new_timeout != KeygenResponseTimeout::<T, I>::get() {
+						KeygenResponseTimeout::<T, I>::put(new_timeout);
+						Self::deposit_event(Event::<T, I>::PalletConfigUpdated { update });
+					}
+				},
 
-			KeygenSlashAmount::<T, I>::put(amount_to_slash);
+				PalletConfigUpdate::KeygenSlashAmount { amount_to_slash } => {
+					if amount_to_slash != KeygenSlashAmount::<T, I>::get() {
+						KeygenSlashAmount::<T, I>::put(amount_to_slash);
+						Self::deposit_event(Event::<T, I>::PalletConfigUpdated { update });
```

Comment:
Then we can pull the event outside of the match statement too (since it can't fail).

Reply:
Sure thing.